sage/pallet/src/lib.rs

```diff
@@ -648,27 +673,40 @@ pub mod pallet {
 			LockedAssets::<T, I>::get(asset_id)
 		}
 
-		fn do_transfer_asset(
+		pub(crate) fn do_transfer_asset(
 			from: &AccountIdOf<T>,
 			to: &AccountIdOf<T>,
-			season_id: &SeasonIdOf<T, I>,
+			asset_season_id: &SeasonIdOf<T, I>,
 			asset_id: &AssetIdOf<T, I>,
 		) -> DispatchResult {
-			let mut from_asset_ids = AssetOwners::<T, I>::get(from, season_id);
-			from_asset_ids.retain(|owned_asset_id| owned_asset_id != asset_id);
+			let technical_account = Self::technical_account_id();
+
+			if from != &technical_account {
+				// The technical account doesn't keep track of the assets transferred to it
+				// so these storage entries are only populated if the the asset is being
+				// transferred from a player account
+				AssetOwners::<T, I>::remove(from, asset_id);
+				AmountAssetsOwned::<T, I>::mutate(from, |owned_count| owned_count.saturating_dec());
+			}
 
```

Comment:
There isn't really a reason for it, when we designed it with Cedric that was the way we deemed it to be. I do prefer this approach vs the asset one, specially since the game developer is in control of the Asset structure and I don't like giving that power to a 3rd party.

Reply:
Ok, but how about just an extra storage map that tracks if an asset is locked? Why do we actually transfer it?

Reply:
Cedric's paranoia that someone could tamper with storage